homeassistant/components/device_tracker/asuswrt.py

```diff
@@ -112,12 +115,17 @@ def __init__(self, config):
         self.ssh_key = config.get('ssh_key', config.get('pub_key', ''))
         self.protocol = config[CONF_PROTOCOL]
         self.mode = config[CONF_MODE]
+        self.port = config.get(CONF_PORT, 22)
+        self.ssh_args = {}
 
         if self.protocol == 'ssh':
+
+            if self.port != DEFAULT_SSH_PORT:
+                self.ssh_args['port'] = self.port
```

Comment:
This is entirely fine, but my take would be to simply always be explicit about the port. That way the code path that you use when specifying an alternate port is always used.